srcs/server/server.cpp

```diff
@@ -128,19 +129,38 @@ namespace ft
 		{
 			httpRequest_pair_map_.erase(deleted_client.client_id);
 		}
+		catch (const ft::Socket::readCGIfd &readCGIexc)
+		{
+			socket_.register_new_client_(readCGIexc.cgi_fd_, true);
+			// client socket fd
+			cgi_client_socket_.first = recieved_msg.client_id;
+			// cgi socket fd
+			cgi_client_socket_.second = readCGIexc.cgi_fd_;
+		}
```

Comment:
This is where cgi_fd is set up in the server. The readCGI exception was thrown from do_cgi function in http_method